src/AspNet.Security.OAuth.Validation/Events/MessageReceivedContext.cs

```diff
@@ -1,11 +1,11 @@
 ﻿using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Http;
 
-namespace AspNet.Security.OAuth.Validation.Events
+namespace AspNet.Security.OAuth.Validation
 {
-    public class ReceivingTokenContext : BaseControlContext
+    public class MessageReceivedContext : BaseControlContext
```

Comment:
I'm for clearer naming and `MessageReceived` is not really saying anything. If I'd get to implement this event, my first mind would be "The message was received. What now?".
Also, as it CAN be used to do many things, what it is MEANT to be used for is token extraction, which `ExtractToken` (or sth like this) defines quite clearly.